recipes/openal/all/conanfile.py

```diff
@@ -30,16 +30,65 @@ def _build_subfolder(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+    @property
+    def _supports_cxx14(self):
+        if self.settings.compiler == "clang" and self.settings.compiler.libcxx in ("libstdc++", "libstdc++11"):
+            if tools.Version(self.settings.compiler.version) < "9":
+                return False, "openal on clang {} cannot be built with stdlibc++(11) c++ runtime".format(self.settings.compiler.version)
+        min_version = {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+        }.get(str(self.settings.compiler))
+        if min_version:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                return False, "This compiler version does not support c++14"
+            else:
+                return True, "Unknown compiler. Assuming your compiler supports c++14"
+        return True, None
+
+    @property
+    def _supports_cxx11(self):
+        if self.settings.compiler == "clang" and self.settings.compiler.libcxx in ("libstdc++", "libstdc++11"):
+            if tools.Version(self.settings.compiler.version) < "9":
+                return False, "openal on clang {} cannot be built with stdlibc++(11) c++ runtime".format(self.settings.compiler.version)
+        min_version = {
+            "Visual Studio": "13",
+            "gcc": "5",
+            "clang": "5",
+        }.get(str(self.settings.compiler))
+        if min_version:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                return False, "This compiler version does not support c++11"
+            else:
+                return True, "Unknown compiler. Assuming your compiler supports c++11"
+        return True, None
+
+    @property
+    def _openal_cxx_backend(self):
+        return tools.Version(self.version) >= "1.20"
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-        del self.settings.compiler.libcxx
+        # OpenAL's API is pure C, thus the c++ standard does not matter
+        # Because the backend is C++, the C++ STL matters
         del self.settings.compiler.cppstd
```

Comment:
But you can still compile openal with a different standard, so compiler.cppstd matters.

Reply:
The cppstd will only change the internals. The external will not change a thing.
I don't think people cares whether openal is compiled with c++14 or c++20.

Keeping libcxx will also requires adding patches removing the hard coding of the c++ standard.
e.g. https://github.com/kcat/openal-soft/blob/d062c16629a93e37366ab88d60d2df3372a234e2/CMakeLists.txt#L153-L155
I don't think this is worth the effort.